src/test/java/org/assertj/core/internal/bytes/Bytes_assertIsEven_Test.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2020 the original author or authors.
+ */
+package org.assertj.core.internal.bytes;
+
+import static java.lang.String.format;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+import static org.assertj.core.test.TestData.someHexInfo;
+import static org.assertj.core.test.TestData.someInfo;
+
+import org.assertj.core.api.AssertionInfo;
+import org.assertj.core.internal.Bytes;
+import org.assertj.core.internal.BytesBaseTest;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Tests for <code>{@link Bytes#assertIsEven(AssertionInfo, Byte)}</code>.
+ *
+ * @author Cal027
+ */
+public class Bytes_assertIsEven_Test extends BytesBaseTest {
```

Comment:
Prefer using `expectAssertionError` over `assertThatExceptionOfType(AssertionError.class)` and introduce GIVEN WHEN THEN as indicated in https://github.com/joel-costigliola/assertj-core/blob/master/CONTRIBUTING.md.
`InputStreams_assertIsEmpty_Test` is a good example to follow too.

Most of these tests can be parameterized (thanks JUnit 5), ex in `Classes_assertHasSuperclass_Test`:
```
  @ParameterizedTest
  @ValueSource(classes = { Object.class, Comparable.class, String.class })
  void should_fail_if_actual_has_not_given_class_as_direct_superclass(Class<?> superclass) {
    // GIVEN
    Class<?> actual = Integer.class;
    // WHEN
    AssertionError assertionError = expectAssertionError(() -> classes.assertHasSuperclass(someInfo(), actual, superclass));
    // THEN
    then(assertionError).hasMessage(shouldHaveSuperclass(actual, superclass).create());
  }
```
You can use short, byte, int and long in `ValueSource`, ex:
```
  @ParameterizedTest
  @ValueSource(bytes = { 0, 2, -2, 0x02, -0x02 })
  void should_succeed_since_actual_is_even(byte b) {
    bytes.assertIsEven(someInfo(), b);
  }
```

Reply:
I'm only commenting this class but apply the same changes to other similar tests.